doc/source/groupby.rst

```diff
@@ -1002,6 +1002,58 @@ See the :ref:`visualization documentation<visualization.box>` for more.
   to ``df.boxplot(by="g")``. See :ref:`here<visualization.box.return>` for
   an explanation.
 
+
+.. _groupby.pipe:
+
+Piping function calls
+~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 0.17.0
+
+Similar to the funcionality provided by ``DataFrames`` and ``Series``, functions that take ``GroupBy`` objects can be chained together using a ``pipe`` method to allow for a cleaner, more readable syntax.
+
+Imagine that one had functions f, g, and h that each takes a ``DataFrameGroupBy`` as well as a single argument and returns a ``DataFrameGroupBy``, and one wanted to apply these functions in succession to a grouped DataFrame.  Instead of having to deeply compose these functions and their arguments, such as:
+
+.. code-block:: python
+
+   >>> h(g(f(df.groupby('group'), arg1), arg2), arg4)
+
+one can write the following:
+
+.. code-block:: python
+
+  >>> (df
+         .groupby('group')
+         .pipe(f, arg1)
+         .pipe(g, arg2)
+         .pipe(h, arg3))
+
+For a more concrete example, imagine a user had a DataFrame with a column 'A' representing a logical group and columns 'B' and 'C' representing numeric values.  Imagine further that the user wanted to group the data by column 'A' and, for each group, the user wanted to take the difference between the maximum value of 'B' in that group and the overal minimum value of 'C' (across all values in all groups).  The user then wanted to continue to transform this data, for example by squaring each resulting value.  One could write this as a pipeline of functions applied to the original dataframe:
+
+.. code-block:: python
+
+    def f(dfgb):
+        """
+        Take a DataFrameGroupBy and return a Series
+        where each value corresponds to the maximum
+        value of column 'B' in each group minus the
+        global minimum of column 'C'.
+        """
+        return dfgb.B.max() - dfgb.C.min().min()
```

Comment:
Not to be too nitpicky, but also this example looks more easy without pipe to me?

```
df.groupby('A')['B'].max() - df['C'].min()
```